numerics/root_finders_body.hpp

```diff
@@ -57,6 +59,102 @@ Argument Bisect(Function f,
   }
 }
 
+// The implementation is translated from the ALGOL 60 in [Bre73], chapter 4,
+// section 6, with the notation adjusted to more closely mirror the formulæ from
+// the preceding sections.
+template<typename Argument, typename Function>
+Argument Brent(Function f,
+               Argument const& lower_bound,
+               Argument const& upper_bound) {
+  using Value = decltype(f(lower_bound));
+  Value const zero{};
+
+  // We do not use std::numeric_limits<double>::epsilon, because it is 2ϵ in
+  // Brent’s notation: Brent uses ϵ = β^(1-τ) / 2 for rounded arithmetic, thus
+  // here ϵ = 2^-τ, see (2.9).
+  constexpr double ϵ = 0x1p-53;
```

Comment:
I would prefer to write `std::numeric_limits<double>::epsilon / 2`, the comment makes it very clear why you do that.  The literal is a code smell.  Alternatively, write `radix ** (1 - digits) / 2` somehow.